widgetssdk/src/main/java/com/glia/widgets/callvisualizer/ActivityWatcherForCallVisualizer.kt

```diff
@@ -128,326 +94,245 @@ class ActivityWatcherForCallVisualizer(
     }
 
-    private fun removeScreenSharingCallback() {
-        screenSharingController.removeViewCallback(screenSharingViewCallback)
+    private fun getConfigurationBuilder(): Configuration.Builder {
+        return Configuration.Builder().setWidgetsConfiguration(Dependencies.getSdkConfigurationManager().createWidgetsConfiguration())
     }
 
-    private fun setupScreenSharingViewCallback(resumedActivity: WeakReference<Activity?>) {
+    override fun showToast(message: String, duration: Int) {
         resumedActivity.get()?.let {
-            screenSharingViewCallback = object : ScreenSharingController.ViewCallback {
-                override fun onScreenSharingRequestError(exception: GliaException?) {
-                    exception?.run { showToast(it, exception.debugMessage) }
-                }
-
-                override fun onScreenSharingStarted() {
-                    if (Glia.isInitialized()) {
-                        serviceChatHeadController.init()
-                    }
-                    val gliaOrRootView: View? = getGliaViewOrRootView(it)
-                    serviceChatHeadController.onResume(gliaOrRootView)
-                }
-            }
+            Toast.makeText(it, message, duration).show()
         }
     }
 
-    @VisibleForTesting
-    fun setupDialogCallback(resumedActivity: WeakReference<Activity?>) {
-        val activity = resumedActivity.get() ?: return
-
-        dialogCallback = DialogController.Callback {
-            when (it.mode) {
-                Dialog.MODE_NONE -> dismissAlertDialog()
-                Dialog.MODE_MEDIA_UPGRADE -> activity.runOnUiThread {
-                    showUpgradeDialog(resumedActivity, it as DialogState.MediaUpgrade)
-                }
-                Dialog.MODE_OVERLAY_PERMISSION -> activity.runOnUiThread {
-                    showOverlayPermissionsDialog(resumedActivity)
-                }
-                Dialog.MODE_START_SCREEN_SHARING -> activity.runOnUiThread {
-                    showScreenSharingDialog(resumedActivity)
-                }
-                Dialog.MODE_ENABLE_NOTIFICATION_CHANNEL -> activity.runOnUiThread {
-                    showAllowNotificationsDialog(resumedActivity)
-                }
-                Dialog.MODE_ENABLE_SCREEN_SHARING_NOTIFICATIONS_AND_START_SHARING ->
-                    activity.runOnUiThread {
-                        showAllowScreenSharingNotificationsAndStartSharingDialog(resumedActivity)
-                    }
-                Dialog.MODE_VISITOR_CODE ->
-                    activity.runOnUiThread {
-                        showVisitorCodeDialog(resumedActivity)
-                    }
-                else -> {
-                    Logger.d(TAG, "Unexpected dialog mode received")
-                }
-            }
+    override fun dismissAlertDialog(manualDismiss: Boolean) {
+        Logger.d(TAG, "Dismiss alert dialog")
+        try {
+            // dialogController.dismissCurrentDialog() // TODO Double dismiss?
```

Comment:
Have you found the reason for the double dismiss or this TODO is still in progress?

Reply:
Still in progress